Test/AwsCommonRuntimeKitTests/http/HttpTests.swift

```diff
@@ -0,0 +1,145 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+import XCTest
+@testable import AwsCommonRuntimeKit
+import AwsCCommon
+
+class HttpTests: CrtXCBaseTestCase {
```

Comment:
This is a first iteration of getting a simple http get request working with memory & lifetime issues fixed. This also helps to visualize how users can make a simple get http request. I will improve these tests coverage (other type of requests) and make helper functions more generic in a future PR. 

There are two tests for two implementations of makeRequest.